### PR TITLE
Roles

### PR DESCRIPTION
**Pull request by ivchenkodima**
http://docs.mithril1.apiary.io/#reference/0/roles

### CODE REVIEW

File: app/common/config.js [hunk 1]
```diff
@@ -9,7 +9,9 @@ if (global.__CLIENT__ && window && window.__CONFIG__) {
 
 export const PORT = config.PORT || process.env.PORT || 8080;
 export const HOSTNAME = typeof window !== 'undefined' ? window.location.origin : (config.HOSTNAME || 'http://localhost:8080');
-export const API_HOST = config.API_HOST || process.env.API_HOST || 'http://private-anon-8b83c175a2-ehealthapi1.apiary-mock.com';
+
+// export const API_HOST = config.API_HOST || process.env.API_HOST || 'https://private-anon-7185b4ec3c-mithril1.apiary-mock.com';
```
Comment (alexeybondarenko) [inline app/common/config.js]: Could be removed

File: app/common/containers/pages/CreateRolePage/redux.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+export const createRole = createAction('roles/CREATE_ROLE');
+
+export const onCreateRole = body => dispatch =>
+  dispatch(fromRoles.createRole(body))
+    .then((action) => {
+      if (action.error) throw action;
+      return Promise.all([
+        dispatch(createRole(action.payload.result)),
+        dispatch(push('/roles')),
```
Comment (alexeybondarenko) [inline app/common/containers/pages/CreateRolePage/redux.js]: Better redirect to `/roles/{created id}`

File: app/common/containers/pages/UpdateRolePage/redux.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import { combineReducers } from 'redux';
+import { push } from 'react-router-redux';
+import { handleAction, createAction } from 'redux-actions';
+import * as fromRoles from 'redux/roles';
+
+export const getRoles = createAction('editRolePage/GET_ROLES');
+export const updateRoles = createAction('editRolePage/UPDATE_ROLES');
+
+export const fetchRoleByID = id => dispatch =>
```
Comment (alexeybondarenko) [inline app/common/containers/pages/UpdateRolePage/redux.js]: Can be replaced with global fromRoles.fetchRoleById. You don't need to store id in the store of the page. You can get ID from URL.

File: app/common/containers/pages/UpdateRolePage/redux.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    if (action.error) throw action;
+    return dispatch(getRoles(action.payload.result));
+  });
+
+export const updateRole = (id, body) => dispatch =>
+  dispatch(fromRoles.updateRole(id, body))
+    .then((action) => {
+      if (action.error) throw action;
+      return Promise.all([
+        dispatch(updateRoles(action.payload.result)),
```
Comment (alexeybondarenko) [inline app/common/containers/pages/UpdateRolePage/redux.js]: Also useless.

File: app/common/redux/roles.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    type: 'roles/FETCH_ROLE_BY_ID_SUCCESS',
+    payload: (action, state, res) => res.json().then(
+      json => normalize(json.data, role)
+    ),
+  }, 'roles/FETCH_ROLE_BY_ID_FAILURE'],
+});
+
+export const createRole = body => invoke({
+  endpoint: `${API_URL}/admin/roles`,
+  method: 'POST',
```
Comment (alexeybondarenko) [inline app/common/redux/roles.js]: Replace with PUT with client-generated uuid V4.

File: app/common/schemas/index.js [hunk 6]
```diff
@@ -10,3 +10,6 @@ export const employee = new schema.Entity('employees');
 export const globalStat = new schema.Entity('globalStat');
```
Comment (alexeybondarenko) [inline app/common/schemas/index.js]: You don't need `globalState, detailsStat, declarationStat`

**Comment by alexeybondarenko** [comment]
Internal Server Error on Heroku Deployment

File: app/common/config.js (hunk 1 above)
Comment (ivchenkodima) [inline app/common/config.js]: +

File: app/common/containers/pages/UpdateRolePage/redux.js (hunk 3 above)
Comment (ivchenkodima) [inline app/common/containers/pages/UpdateRolePage/redux.js]: done

File: app/common/containers/pages/UpdateRolePage/redux.js (hunk 4 above)
Comment (ivchenkodima) [inline app/common/containers/pages/UpdateRolePage/redux.js]: done

File: app/common/schemas/index.js (hunk 6 above)
Comment (ivchenkodima) [inline app/common/schemas/index.js]: done

File: app/common/containers/pages/CreateRolePage/redux.js (hunk 2 above)
Comment (ivchenkodima) [inline app/common/containers/pages/CreateRolePage/redux.js]: done

**Pull request merged**